crypto/pem/pem_lib.c

```diff
@@ -849,7 +850,6 @@ static int get_header_and_data(BIO *bp, BIO **header, BIO **data, char *name,
         /* Check for end of stream (which means there is no header). */
         if (strncmp(linebuf, endstr, ENDLEN) == 0) {
             p = linebuf + ENDLEN;
-            namelen = strlen(name);
```

Comment:
I think there is no need to move this outside since this `if` branch will finally reach to a `break`, which can break out of the loop.